src/evaluate/inspect.py

```diff
@@ -31,41 +32,67 @@ class SplitsNotFoundError(ValueError):
     pass
 
 
-def list_metrics(with_community_metrics=True, with_details=False):
+def list_evaluation_modules(type=None, include_community=True, with_details=False):
     """List all the metrics script available on the Hugging Face Hub.
 
     Args:
-        with_community_metrics (:obj:`bool`, optional, default ``True``): Include the community provided metrics.
-        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the short name.
+        type (:obj:`string`, optional, default ``None``): Type of evaluation modules to list. If ``None`` all types are listed.
+        include_community (:obj:`bool`, optional, default ``True``): Include community modules in the list.
+        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the ID.
+
+    Returns:
+        `list`
     """
-    metrics = huggingface_hub.list_metrics()
-    if not with_community_metrics:
-        metrics = [metric for metric in metrics if "/" not in metric.id]
-    if not with_details:
-        metrics = [metric.id for metric in metrics]
-    return metrics
+    if type is None:
+        evaluations_list = []
+        for type in ["metric", "comparison", "measurement"]:
```

Comment:
Since there is no validation of `type`, a user could write `metrics` instead and might get confused by not getting anything back.